### DISCUSSION TITLE
Nova.booted callback functionality only partially implemented

### DISCUSSION THREAD

**Discussion by ianrobertsFF** [laravel/nova-issues#4180] [Ideas & Feature Requests]
- Laravel Version: 9.11.0
- Nova Version: 4.4.1
- PHP Version: 8.1.4
- Database Driver & Version: N/A
- Operating System and Version: N/A
- Browser type and version: N/A
- Reproduction Repository: N/A

### Description:

Unsure if this should be a bug report or a discussion, erred on the side of a bug report as the commented out code made me think that this was an oversight, rather than a feature request. 

In app.js the `booted` functionality appears to be only partially implemented and doesn't act in the expected way. `booting` callbacks run at the start of `liftOff`, expected functionality would be that `booted` callbacks run at the end of `liftOff`. I'm not sure if this was originally the intended behaviour as pushing the callback into the `bootedCallbacks` array has been commented out:

```
booted(callback) {
    // this.bootedCallbacks.push(callback)
    callback(this.app, this.store)
  }
```

### Detailed steps to reproduce the issue on a fresh Nova installation:

N/A

**Comment by ianrobertsFF** [comment]
Okay, so apparently this is a feature request, despite the janky current behaviour.

As it is a feature request, could we please have consistent behaviour for booted in a future release?

Thanks

**Reply by crynobone** [reply]
All scripts should be registered before `liftOff` so the request doesn't have any affect. 

Which is why the particular line was commented in Nova 3 and Nova 4

**Reply by ianrobertsFF** [reply]
It has the effect of solving this issue: https://github.com/laravel/nova-issues/issues/4143

boot() makes sense where it is, booted at the end resolves this issue.

**Reply by crynobone** [reply]
That a different issue and solution all together. If that's all you need then there's no need for this request.

**Reply by ianrobertsFF** [reply]
It's not all I need, but you stating that it doesnt have any effect is demonstrably false, as there is clearly a need to overwrite things during Nova's liftoff.

If booted was properly implemented rather than its current and ineffective form this issue wouldn't even exist.

**Reply by crynobone** [reply]
This suggested changes has zero affect to the linked issue.

**Reply by ianrobertsFF** [reply]
Okay mate, forget it, clearly you don't want to discuss it. You've declared it "no affect" and thats the end of it.

**Comment by crynobone** [comment]
This has been fixed in 5.2.0
